dask/bag/tests/test_random.py

```diff
@@ -0,0 +1,97 @@
+import pytest
+
+import dask.bag as db
+from dask.bag import random
+
+
+def test_choices_size():
+    """
+    Number of randomly sampled elements are exactly k.
+    """
+    a = db.from_sequence(range(20), npartitions=3)
+    s = random.choices(a, k=2)
+    assert len(list(s.compute())) == 2
+
+
+def test_choices_size_over():
+    """
+    Number of randomly sampled are more than the elements.
+    """
+    a = db.from_sequence(range(3), npartitions=3)
+    s = random.choices(a, k=4)
+    assert len(list(s.compute())) == 4
+
+
+def test_choices_size_over_repartition():
+    """
+    Number of randomly sampled are more than the elements on each partition.
+    """
+    a = db.from_sequence(range(10), partition_size=9)
+    a = a.repartition(3)
+    s = random.choices(a, k=2)
+    assert len(list(s.compute())) == 2
+
+
+def test_choices_size_over_perpartition():
+    """
+    Number of randomly sampled are more than the elements of a partition.
+    """
+    a = db.from_sequence(range(10), partition_size=9)
+    s = random.choices(a, k=2)
+    assert len(list(s.compute())) == 2
+
+
+def test_choices_size_over_two_perpartition():
+    """
+    Number of randomly sampled are more than the elements of two partitions.
+    """
+    a = db.from_sequence(range(10), partition_size=9)
+    s = random.choices(a, k=10)
+    assert len(list(s.compute())) == 10
+
+
+def test_sample_size():
+    """
+    Number of randomly sampled elements are exactly k.
+    """
+    a = db.from_sequence(range(20), npartitions=3)
+    s = random.sample(a, k=2)
+    assert len(list(s.compute())) == 2
+
+
+def test_sample_size_over():
+    """
+    Number of randomly sampled are more than the elements.
+    """
+    a = db.from_sequence(range(3), npartitions=3)
+    s = random.sample(a, k=4)
+    with pytest.raises(ValueError):
+        s.compute()
+
+
+def test_sample_size_over_repartition():
+    """
+    Number of randomly sampled are more than the elements on each partition.
+    """
+    a = db.from_sequence(range(10), partition_size=9)
+    a = a.repartition(3)
+    s = random.sample(a, k=2)
+    assert len(list(s.compute())) == 2
+
+
+def test_sample_size_over_perpartition():
+    """
+    Number of randomly sampled are more than the elements of a partition.
+    """
+    a = db.from_sequence(range(10), partition_size=9)
+    s = random.sample(a, k=2)
+    assert len(list(s.compute())) == 2
+
+
+def test_sample_size_over_two_perpartition():
+    """
+    Number of randomly sampled are more than the elements of two partitions.
+    """
+    a = db.from_sequence(range(10), partition_size=9)
+    s = random.sample(a, k=10)
+    assert len(list(s.compute())) == 10
```

Comment:
You should have some tests for the actual contents of the returned values. At least to check that there is replacement or not depending on whether using `choices` or `sample`.

Reply:
I removed the sample method, right now. I had some troubles on testing the exact output from the choices method because of reproducibility problems. 
I evaluated the possibility to add the seed function to bag.random: in order to mimic python random seed. 
It could be done as a further development, but is not super straightforward since there are issues that must be resolved with it.

Reply:
Yeah that makes sense. I just meant something like asserting that all the values in the output are in the input bag.

Reply:
Ok, I just added more specific tests on latest commit. Thanks for the advice